packages/beachball/src/publish/publishToRegistry.ts

```diff
@@ -42,7 +35,7 @@ export async function publishToRegistry(originalBumpInfo: BumpInfo, options: Bea
 
   const packagesToPublish = toposortPackages([...modifiedPackages, ...newPackages], packageInfos);
 
-  packagesToPublish.forEach(pkg => {
+  for (const pkg of packagesToPublish) {
```

Comment:
let's do all the prepublish hooks for all packages in one loop, and then loop thru the packages again to publish. If we do this in phases, then we can catch prepub hook errors and prevent any publishes. partial publishes is SO BAD.

Reply:
Great idea!  Updated.